wunderio/next4drupal-project
# NEX-12: elasticsearch

This PR adds elasticsearch search functionality to the site, both in the drupal backend and in the next frontend. It's a big one! :) 

### Backend

* Added elasticsearch to lando
* added elasticsearch helper (7.0 version)
* added custom module with: 
  * index definitions
  * normalizer
  * proxy to make actual searches
  
### Frontend

The frontend work is based on the existing https://github.com/wunderio/nextjs-elastic-search-ui-example effort

* Added latest version of https://github.com/elastic/search-ui
* Overridden ui components to be able to translate the ui strings, and use tailwind css to make the ui consistent, and show how the overriding can be done
* added facets and query matching the schema and content 
* added search page and search bar in the navbar in all pages
* added wunder colors

The "frontend of the frontend" implementation (classes, search bar in navbar etc) should be improved particularly in terms of responsiveness, but I propose to make that an independent jira story.


### How to test

1. set up the project from scratch, by following carefully the README file, which now includes also a step to create elasticsearch indices. (run lando rebuild if you have the project running already, since this adds elasticsearch to lando)
2. log into the site, create content and translate it. Add tags to the article content type
3. access `https://frontend.lndo.site/search`
* click the search button and see all results
* search for a keyword and see results
* input a search result that won't find any result and see the message about no search results found

Switch to a different language and navigate to /search



### Known limitations

* when clicking on the language switcher while on the /search page, you are taken to the frontpage instead of the translated search page, this is due to how the language switcher works. For now we decided to go with this.



### Screenshots

![image](https://user-images.githubusercontent.com/185412/212025379-9659b16d-1a1b-4f06-b9f6-dfb1cbeefa43.png)
